tensorflow/lite/experimental/micro/kernels/cmsis-nn/depthwise_conv.cc

```diff
@@ -117,12 +128,114 @@ void EvalFloat(TfLiteContext* context, TfLiteNode* node,
+                             const TfLiteTensor* bias, TfLiteTensor* output) {
+#if defined(ARM_MATH_DSP) && defined(ARM_MATH_LOOPUNROLL)
+    DepthwiseParams op_params;
+    op_params.padding_type = PaddingType::kSame;
+    op_params.padding_values.width = data->padding.width;
+    op_params.padding_values.height = data->padding.height;
+    op_params.stride_width = params->stride_width;
+    op_params.stride_height = params->stride_height;
+    op_params.dilation_width_factor = params->dilation_width_factor;
+    op_params.dilation_height_factor = params->dilation_height_factor;
+    op_params.depth_multiplier = params->depth_multiplier;
+    op_params.input_offset = -input->params.zero_point;
+    op_params.weights_offset = 0;
+    op_params.output_offset = output->params.zero_point;
+    // TODO(b/130439627): Use calculated value for clamping.
+    op_params.quantized_activation_min = std::numeric_limits<int8_t>::min();
+    op_params.quantized_activation_max = std::numeric_limits<int8_t>::max();
+    RuntimeShape filter_shape = GetTensorShape(filter);
+    const int filter_height = filter_shape.Dims(1);
+    const int filter_width = filter_shape.Dims(2);
+    RuntimeShape input_shape = GetTensorShape(input);
+    const int input_height = input_shape.Dims(1);
+    const int input_width = input_shape.Dims(2);
+    const int input_depth = input_shape.Dims(3);
+    RuntimeShape output_shape = GetTensorShape(output);
+    const int output_height = output_shape.Dims(1);
+    const int output_width = output_shape.Dims(2);
+    RuntimeShape bias_shape = GetTensorShape(bias);
+
+    if (op_params.depth_multiplier == 1) {
+      int16_t* buf = nullptr;
+      const int32_t buf_size =
+        arm_depthwise_conv_s8_opt_get_buffer_size(input_depth,
+                                                  filter_width,
+                                                  filter_height);
+      if (get_cmsis_scratch_buffer(context, &buf, buf_size) != kTfLiteOk) {
+        return kTfLiteError;
+      }
+      if (arm_depthwise_conv_s8_opt(
+            GetTensorData<int8_t>(input),
+            input_width, input_height, input_depth,
+            GetTensorData<int8_t>(filter),
+            input_depth,
+            filter_width, filter_height,
+            op_params.padding_values.width,
+            op_params.padding_values.height,
+            op_params.stride_width,
+            op_params.stride_height,
+            GetTensorData<int32>(bias),
+            GetTensorData<int8_t>(output),
+            data->per_channel_output_shift,
+            data->per_channel_output_multiplier,
+            output_width,
+            output_height,
+            op_params.output_offset,
+            op_params.input_offset,
+            op_params.quantized_activation_min,
+            op_params.quantized_activation_max,
+            op_params.dilation_width_factor,
+            op_params.dilation_height_factor,
+            buf) != ARM_MATH_SUCCESS) {
```

Comment:
You could use TF_LITE_ENSURE_EQ-macro here instead an if. Same for line 202.

Reply:
On a second thought, this one returns arm_status. So I can't use TF_LITE_ENSURE_EQ.

Reply:
ah, I see now. I didn't see the TF_LITE_ENSURE_EQ(). Will fix.